packages/nuxt/src/pages/runtime/router.options.ts

```diff
@@ -43,12 +43,8 @@ export default <RouterConfig> {
     const hasTransition = (route: RouteLocationNormalized) => !!(route.meta.pageTransition ?? defaultPageTransition)
     const hookToWait = (hasTransition(from) && hasTransition(to)) ? 'page:transition:finish' : 'page:finish'
```

Comment:
I think this might be a better default as it handles nested routes

```suggestion
    const hookToWait = (hasTransition(from) && hasTransition(to)) ? 'page:transition:finish' : 'page:loading:end'
```

Reply:
@dennisadriaans you missed the comment above this line ;)

Reply:
@matsn0w Did I? The code checks the transition before selecting the actual hook on line 44, right?